types/lodash/lodash-tests.ts

```diff
@@ -5068,7 +5062,13 @@ namespace TestSome {
         result = _.some<SampleObject>(list, {a: 42});
 
         result = _.some<SampleObject>(dictionary);
-        result = _.some<SampleObject>(dictionary, dictionaryIterator);
+        result = _.some<SampleObject>(numericDictionary, dictionaryIterator);
+        result = _.some<SampleObject>(dictionary, (value, key, collection) => {
```

Comment:
Here is the one thing I added to the tests: a check to make sure the type inference works correctly in strictNullChecks mode (so the type of `key` should be inferred as `string`, not `string | undefined`)